src/include/miopen/solver/implicitgemm_ck_util.hpp

```diff
@@ -632,8 +656,16 @@ ConvSolution InitInvokerFactoryNCHW(const ExecutionContext& ctx,
     auto ck_args = CKArgsType{problem};
 
     auto conv_ptrs = DeviceOpType::GetInstances();
-    auto ptr_iter  = FindConvPtrByID(conv_ptrs, kernel_id);
 
+    std::shared_ptr<CKBWDWeightBufferDescriptor> _ck_buff_des;
+
+    if(problem.GetSpatialDims() == 3 && problem.IsDirectionBackwardWrW())
+    {
+        _ck_buff_des =
```

Comment:
why does a simple struct like this need to be a `shared_ptr`? Let's see if we can avoid it. You can for example use `std::optional<CK...>` and that does not allocate anything on the heap (just an object + a bool).